src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -390,6 +390,53 @@ public boolean isColumnRestrictedByEq(ColumnMetadata columnDef)
                            .anyMatch(p -> ((SingleRestriction) p).isEQ());
     }
 
+    /**
+     * This method determines whether a specified column is restricted on equality or something equivalent, like IN.
+     * It can be used in conjunction with the columns selected by a query to determine which of those columns is 
+     * already bound by the client and not necessarily retrieved by the database.
+     *
+     * @param column a column specification from the same table these restrictions are against
+     *
+     * @return true if the given column is restricted on equality
+     */
+    public boolean isEqualityRestricted(ColumnSpecification column)
```

Comment:
Here you're potentially traversing all columns each on each call. Given you're on a hot path, could you be feeding instead this method with `ColumnMetada`, pulled from the restrictions, and relaying on `KIND` spare a couple loops?

Reply:
Let me mull this one over. In the meantime though, note that we can disable all of this w/ `client_request_size_metrics_enabled` in the YAML.

Reply:
You can disable it yes, but when enabled you want it as performant as possible imo.